src/workerd/jsg/struct.h

```diff
@@ -118,10 +118,19 @@ class StructWrapper<Self, T, TypeTuple<FieldWrappers...>, kj::_::Indexes<indices
     //   it prescribes lexicographically-ordered member initialization, with base members ordered
     //   before derived members. Objects with mutating getters might be broken by this, but it
     //   doesn't seem worth fixing absent a compelling use case.
-
-    return T {
+    auto t = T {
       kj::get<indices>(fields).unwrap(static_cast<Self&>(*this), isolate, context, in)...
     };
+
+    // Note that if a `validate` function is provided, then it will be called after the struct is
+    // unwrapped from v8. This would be an appropriate time to throw an error.
+    // Signature: void validate(jsg::Lock& js);
+    if constexpr (requires { t.validate(); }) {
```

Comment:
should the ` requires { t.validate(); }` here include the `jsg::Lock&` argument?

Reply:
I think it should. That is a good point.

Reply:
Hmm... why didn't this lead to a test failure?

Reply:
When you don't have an argument it just assumes you will take `void validate(any...);